presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -126,13 +129,58 @@ public Block readBlock(ColumnDescriptor columnDescriptor, int vectorSize, Type t
             throws IOException
     {
         ParquetColumnReader columnReader;
-        if (columnReadersMap.containsKey(columnDescriptor)) {
-            columnReader = columnReadersMap.get(columnDescriptor);
+        ColumnIndex columnIndex = new ColumnIndex(currentRowGroup, columnDescriptor);
+        if (columnReadersMap.containsKey(columnIndex)) {
+            columnReader = columnReadersMap.get(columnIndex);
         }
         else {
             columnReader = new ParquetColumnReader(columnDescriptor, fileReader.readColumn(columnDescriptor));
-            columnReadersMap.put(columnDescriptor, columnReader);
+            columnReadersMap.put(columnIndex, columnReader);
         }
         return columnReader.readBlock(vectorSize, type);
     }
+
+    private final class ColumnIndex
+    {
+        private final long rowGroupNumber;
+        private final ColumnDescriptor columnDescriptor;
+
+        public ColumnIndex(long rowGroupNumber, ColumnDescriptor columnDescriptor)
+        {
+            this.rowGroupNumber = rowGroupNumber;
+            this.columnDescriptor = columnDescriptor;
+        }
+
+        public long getRowGroupNumber()
```

Comment:
If these aren't called, just remove them for now.  We can always add them back later